packages/atomic-hosted-page/src/components/atomic-hosted-ui/hosted-ui.ts

```diff
@@ -0,0 +1,87 @@
+interface HostedPage {
+  /**
+   * The HTML markup of the hosted page.
+   */
+  html: string;
+  /**
+   * The JavaScript resources appended to the header of a specific hosted page in the target Coveo Cloud organization.
+   */
+  javascript?: HostedPageJavascriptFile[];
+  /**
+   * The CSS resources appended in the header, pertaining to a specific hosted page in the target Coveo Cloud organization.
+   */
+  css?: HostedPageCSS[];
+  w;
+}
+
+type HostedPageCSS = HostedPageCssInlineFile | HostedPageCssUrlFile;
+
+interface HostedPageCssUrlFile {
+  /**
+   * The URL of CSS stylesheet.
+   */
+  url: string;
+}
+
+interface HostedPageCssInlineFile {
+  /**
+   * The content of the header `style` tag.
+   */
+  inlineContent: string;
+}
+
+interface HostedPageJavascriptFile {
+  /**
+   * Whether the inline code should be treated as a JavaScript module. If this property is `true`, the `type` property will be set to `module` on the `script` tag.
+   */
+  isModule: boolean;
+  /**
+   * The content of the header `script` tag. If this property is defined, the `url` property should not be specified.
+   */
+  inlineContent?: string;
+  /**
+   * The URL of the JavaScript source file. If this property is defined, the `inlineContent` property should not be specified.
+   */
+  url?: string;
+}
+
+export function processHostedPage(
+  element: HTMLElement,
+  hostedPage: HostedPage
+) {
+  element.innerHTML = hostedPage.html;
+  hostedPage.javascript?.forEach((file) => insertJS(file));
+  hostedPage.css?.forEach((file) =>
+    'url' in file ? insertCSSUrl(file) : insertCSSInline(file)
+  );
+}
+function insertJS(file: HostedPageJavascriptFile) {
+  const script = document.createElement('script');
+  if (file.isModule) {
+    script.type = 'module';
+  }
+
+  if (file.url) {
```

Comment:
Since `file.url` and `file.inlineContent` are mutually exclusive, perhaps we should log a warning if they've both been specified.

Reply:
It's unlikely to happen, as it come from our backend, but let's imagine we can code bugs.

What is a good code pattern here is to add a guard/precondition https://en.wikipedia.org/wiki/Guard_(computer_science)

So in essence, you can add a if(fileHasUrlAndInlineContent), then log a warning and early return empty

Reply:
@louis-bompart so you are suggesting we don't insert the JS if there's both a url and inline content, instead of just adding one of them?

Reply:
Yep!

Reply:
> but let's imagine we can code bugs.

😂